### PR TITLE
Updated theme logic

### PR DESCRIPTION
**Pull request by emalsidog**
I am sorry. There are some changes I made yesterday in the Thread component. I have not known that you started working on Front end too. I have removed my today's changes.

About themes. I have left old theme object structure in compatibility purposes but I have also created a new one based on names that I have seen on Figma. Also, I have decided to use redux to store current theme because the ThemeSwitcher can be called anywhere and useTheme hook will not update theme object in the Theme component.

Again, I apologize for my yesterday's changes.

### CODE REVIEW

File: client/src/components/Projects/Thread/index.js [hunk 1]
```diff
@@ -647,18 +727,12 @@ export const Thread = (props) => {
...
-          {t('Copy permalink')}
-        </MenuItem>
-        <MenuItem onClick={handleSelectMessage}>
-          {t('Select message')}
-        </MenuItem>
+        <MenuItem onClick={handleCopyPermalink}>{t('Copy permalink')}</MenuItem>
+        <MenuItem onClick={handleSelectMessage}>{t('Select message')}</MenuItem>
         <MenuItem onClick={handleNewConversationAnchor}>
           {t('New conversation')}
         </MenuItem>
```
Comment (bubuzavrik) [inline client/src/components/Projects/Thread/index.js:738]: You want split this component on some lesser components, cause here too much rows and it's very hard to debug

**Review by bubuzavrik — CHANGES_REQUESTED** [review]
Try to make components 200 rows and lesser, cause if there are 700+, it's very hard to debug

**Comment by emalsidog** [comment]
> Try to make components 200 rows and lesser, cause if there are 700+, it's very hard to debug

Should I make a pull?

**Comment by bubuzavrik** [comment]
> > Try to make components 200 rows and lesser, cause if there are 700+, it's very hard to debug
> 
> Should I make a pull?

Yes

**Review by bubuzavrik — APPROVED** [review]

**Pull request merged**